model/atmosphere/diffusion/src/icon4py/model/atmosphere/diffusion/diffusion.py

```diff
@@ -599,12 +612,13 @@ def _do_diffusion_step(
             smag_offset:
 
         """
-        num_levels = self.grid.num_levels
         # dtime dependent: enh_smag_factor,
-        scale_k(self.enh_smag_fac, dtime, self.diff_multfac_smag, offset_provider={})
+        scale_k.with_connectivities(self.compile_time_connectivities)(
```

Comment:
That is a gt4py question really: 
By using `with_connectivities()` is it then still necessary to add `offset_provider` kwarg, or can that be removed? To me it looks a bit redundant, but it is probably needed to pass the actual runtime connectivity tables?

Also what is the advantage or disadvantage of using it? Using the `with_backend` call it used to be significantly slower than setting a `backend` `@program` decorator, one of the reasons of the backend mess that we have in icon4py...
Maybe that changed as well. 

Maybe we can have a discussion on this with somebody from the gt4py team?

Reply:
same as above.